source/Calamari.AzureServiceFabric/Integration/AzureServiceFabricPowerShellContext.cs

```diff
@@ -57,31 +61,39 @@ public CommandResult ExecuteScript(Script script,
             variables.Set("OctopusFabricTargetScript", script.File);
             variables.Set("OctopusFabricTargetScriptParameters", script.Parameters);
 
-            // Azure PS modules are required for looking up Azure environments (needed for AAD url lookup in Service Fabric world).
-            SetAzureModulesLoadingMethod();
+            var serverCertThumbprint = variables.Get(SpecialVariables.Action.ServiceFabric.ServerCertThumbprint);
+            var connectionEndpoint = variables.Get(SpecialVariables.Action.ServiceFabric.ConnectionEndpoint);
+            var aadUserCredUsername = variables.Get(SpecialVariables.Action.ServiceFabric.AadUserCredentialUsername);
+            var aadUserCredPassword = variables.Get(SpecialVariables.Action.ServiceFabric.AadUserCredentialPassword);
 
             // Read thumbprint from our client cert variable (if applicable).
             var securityMode = variables.Get(SpecialVariables.Action.ServiceFabric.SecurityMode);
             var clientCertThumbprint = string.Empty;
             if (securityMode == AzureServiceFabricSecurityMode.SecureClientCertificate.ToString())
             {
-                var certificateVariable = GetMandatoryVariable(SpecialVariables.Action.ServiceFabric.ClientCertVariable);
+                var certificateVariable = variables.GetMandatoryVariable(SpecialVariables.Action.ServiceFabric.ClientCertVariable);
                 clientCertThumbprint = variables.Get($"{certificateVariable}.{CertificateVariables.Properties.Thumbprint}");
             }
+            else if (securityMode == AzureServiceFabricSecurityMode.SecureAzureAD.ToString())
+            {
+                if (string.IsNullOrWhiteSpace(aadUserCredUsername))  return new CommandResult("Failed to find a value for the AAD username.", 0);
+                if (string.IsNullOrWhiteSpace(aadUserCredPassword))  return new CommandResult("Failed to find a value for the AAD password.", 0);
```

Comment:
This validation was originally performed on line 74-81 of `AzureServiceFabricContext.ps1`
Now that we're getting the AAD token in C#, we need to validate it in this file instead.